.github/workflows/release-php.yaml

```diff
@@ -0,0 +1,67 @@
+        # This if could be moved to the parent `job` section when it's
+        # supported by GitHub (https://github.com/community/community/discussions/37883)
+        if: github.event.ref == format('refs/tags/php/{0}', matrix.versions.php)
+        shell: bash
+        run: |
+          sudo mv php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi{,.wasm}
+      - name: Setup binaryen
+        # This if could be moved to the parent `job` section when it's
+        # supported by GitHub (https://github.com/community/community/discussions/37883)
+        if: github.event.ref == format('refs/tags/php/{0}', matrix.versions.php)
+        shell: bash
+        run: |
+          wget https://github.com/WebAssembly/binaryen/releases/download/version_${{ env.BINARYEN_VERSION }}/binaryen-version_${{ env.BINARYEN_VERSION }}-x86_64-linux.tar.gz
+          tar -xf binaryen-version_${{ env.BINARYEN_VERSION }}-x86_64-linux.tar.gz --strip-components=1 -C /opt
+          rm binaryen-version_${{ env.BINARYEN_VERSION }}-x86_64-linux.tar.gz
+      - name: Optimize release artifacts
+        # This if could be moved to the parent `job` section when it's
+        # supported by GitHub (https://github.com/community/community/discussions/37883)
+        if: github.event.ref == format('refs/tags/php/{0}', matrix.versions.php)
+        shell: bash
+        run: |
+          sudo /opt/bin/wasm-opt -Os -o php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi.size-optimized.wasm php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi.wasm
+          sudo /opt/bin/wasm-opt -O -o php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi.speed-optimized.wasm php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi.wasm
+      - name: Append version to release artifacts
+        if: github.event.ref == format('refs/tags/php/{0}', matrix.versions.php)
+        run: |
+          sudo mv php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi{,-${{ matrix.versions.php }}}.wasm
+          sudo mv php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi{,-${{ matrix.versions.php }}}.size-optimized.wasm
+          sudo mv php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi{,-${{ matrix.versions.php }}}.speed-optimized.wasm
+      - name: Release
+        # This if could be moved to the parent `job` section when it's
+        # supported by GitHub (https://github.com/community/community/discussions/37883)
+        if: github.event.ref == format('refs/tags/php/{0}', matrix.versions.php)
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi-${{ matrix.versions.php }}.wasm
+            php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi-${{ matrix.versions.php }}.size-optimized.wasm
+            php/build-output/php/php-${{ matrix.versions.php }}/bin/php-cgi-${{ matrix.versions.php }}.speed-optimized.wasm
```

Comment:
This job has the `fail-fast` policy set to `false`. I believe this means steps will continue even when some of them fails. Shall we protect this one to be run only if previous were successful? I'm not sure if the release will be created even if there's no file inside. I'm trying to avoid empty releases when something fails.

Reply:
`fail-fast` acts at the job level. Steps are still being aborted if a previous step fails. By default, when a job (list of steps) fail, the rest of the jobs that are being parallelised are aborted if a job has failed. With `fail-fast: false`, we execute all jobs nevertheless, even if a parallel job has failed its execution.

In build pipelines I think `fail-fast: false` is valuable, because a run might provide info about a couple of failures, whereas with `fail-fast: true` we would only get information about the first failing job, fix it, and then find a second failure on a subsequent run. I think is better to find out all at once.

With release pipelines I think an all or nothing approach is preferred. I would like to improve the release pipeline so that we have this all or nothing approach, and the GitHub release is only created (and filled with artifacts) when all artifacts are available. This way we can't end up with GitHub releases that are not populated with all artifacts that we produce.

If you are fine with it, I am going to open an issue to keep track of that and improve the pipeline later on.

Reply:
Exactly! I agree with you. Since this is a `release` pipeline, I agree that having `fail-fast` to `false` is preferred. However, this case is a bit special as only a single job will be run. All steps will be skipped for different tag versions.

Taking into account other jobs won't have any step due to the `if` conditionals, I'm fine moving forward with this and tracking it in a separate issue 👍

Reply:
Updated, thanks :) Not creating the issue right now because as you said only one job will run (although because of GitHub limitations only steps can be filtered out, but only one job will really run its steps)